### PR TITLE
Fixed firebase-auth tests and issue with promises always resolving to then

### LINKED ISSUE #13: _handleSignIn causes the returned Promise to always evaluate to then
The `catch` in the signInWithPopup usage example isn't ever hit due to the promise being handled in:

https://github.com/firebase/polymerfire/blob/master/firebase-auth.html#L213

The best solution is to throw the error after firing off the event that way the end-user can handle it as well.

### PR DESCRIPTION
**Pull request by nick-woodward**
Fixes #13

@cdata I noticed that authElement.user doesn't get set right away. I'm guessing this is due to it waiting for `__authChanged` to fire.

### CODE REVIEW

File: test/firebase-auth.html [hunk 1]
```diff
@@ -50,8 +50,18 @@
         });
 
         test('eventually signs in', function() {
-          return authElement.signInAnonymously().then(function() {
-            expect(authElement.user).to.be.ok;
+          return authElement.signInAnonymously().then(function(user) {
+            expect(user.uid).to.be.ok;
+          }).catch(function() {
+            throw 'Sign in Anonymously should not have failed';
```
Comment (cdata) [inline test/firebase-auth.html]: This `throw` is unnecessary. Mocha will catch a rejected Promise automatically.

**Comment by cdata** [comment]
This looks good to me, thanks for "catching" this bug.

I left one minor piece of feedback about the test, please update and then we can merge.

**Comment by nick-woodward** [comment]
The `throw` has been removed as "promised".

**Comment by cdata** [comment]
LGTM 👍 Thanks!

**Pull request merged**